unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -1192,6 +1192,14 @@ resolveRootBranchHashV2 mayRoot = case mayRoot of
     h <- expandShortCausalHash sch
     lift (resolveCausalHashV2 (Just h))
 
+normaliseRootCausalHash :: Maybe (Either ShortCausalHash CausalHash) -> Backend Sqlite.Transaction (V2Branch.CausalBranch Sqlite.Transaction)
```

Comment:
The local server passes ShortCausalHashes if provided by a query param.

Enlil passes full hashes resolved from the Branch Head;

That's why we tag the args as either Left or Right.